samcli/commands/_utils/options.py

```diff
@@ -200,6 +200,23 @@ def resolve_s3_callback(ctx, param, provided_value, artifact, exc_set, exc_not_s
     return provided_value
 
 
+def skip_prepare_iac_callback(ctx: click.core.Context, param: click.Option, provided_value: bool):
+    """
+    Callback for --skip-prepare-iac to check if --hook-package-id is also specified
+
+    Parameters
+    ----------
+    ctx: click.core.Context
+        Click context
+    param: click.Option
+        Parameter properties
+    provided_value: bool
+        True if option was provided
+    """
+    if provided_value and not ctx.params.get("hook_package_id", None):
```

Comment:
also check in `ctx.default_map` .. so we also cover the case if the customer added this option to samconfig toml file